ouroboros-network/lib/Ouroboros/Network/PeerSelection/Governor/EstablishedPeers.hs

```diff
@@ -825,25 +826,22 @@ aboveTargetBigLedgerPeers actions@PeerSelectionActions {
         -- pick active peer.  The `min` is taken so that `pickPeers` is given
         -- consistent number of peers with the set of peers available to demote,
         -- i.e. `availableToDemote`.
-        numBigLedgerPeersToDemote    = min ( numEstablishedBigLedgerPeers
-                                           - targetNumberOfEstablishedBigLedgerPeers)
-                                           ( numEstablishedBigLedgerPeers
-                                           - numActiveBigLedgerPeers)
-                                     - Set.size inProgressDemoteWarm
-                                     - Set.size inProgressPromoteWarm
+        numBigLedgerPeersToDemote    =
+            min (numEstablishedBigLedgerPeers - targetNumberOfEstablishedBigLedgerPeers)
+                (numEstablishedBigLedgerPeers - numActiveBigLedgerPeers)
+          - Set.size inProgressDemoteWarm
+          - Set.size inProgressPromoteWarm
+          - Set.size (Set.intersection warmBigLedgerPeers inProgressDemoteToCold)
```

Comment:
I first thought that we're subtracting `inProgressDemoteWarm` twice, but it isn't, because async demotions are only in `inProgressDemoteToCold`, isn't it?

Yes, that's right: I checked that `assertPeerSelectionState` states that `inProgressDemoteWarm`, `inProgressPromoteWarm` and `inProgressDemoteToCold` are all disjoint.

It would be good to extend the comment.